### PR TITLE
Replace Base64 from Apache Commons Codec with equivalent from Guava.

### PR DESCRIPTION
**Pull request by mattwhisenhunt**
@saicheems

### CODE REVIEW

**Comment by mattwhisenhunt** [comment]
I signed it!

**Comment by saicheems** [comment]
@garrettjonesgoogle requesting review

File: google-http-client/src/main/java/com/google/api/client/util/Base64.java [hunk 1]
```diff
@@ -36,35 +36,41 @@
    * @param binaryData binary data to encode or {@code null} for {@code null} result
    * @return byte[] containing Base64 characters in their UTF-8 representation or {@code null} for
    *         {@code null} input
-   * @see org.apache.commons.codec.binary.Base64#encodeBase64(byte[])
    */
   public static byte[] encodeBase64(byte[] binaryData) {
-    return org.apache.commons.codec.binary.Base64.encodeBase64(binaryData);
+    if(binaryData == null){
```
Comment (saicheems) [inline google-http-client/src/main/java/com/google/api/client/util/Base64.java]: minor nit: spaces between `(` and `{`

File: google-http-client/src/main/java/com/google/api/client/util/Base64.java [hunk 2]
```diff
@@ -36,35 +36,41 @@
...
    * @return byte[] containing Base64 characters in their UTF-8 representation or {@code null} for
    *         {@code null} input
-   * @see org.apache.commons.codec.binary.Base64#encodeBase64URLSafe(byte[])
    */
   public static byte[] encodeBase64URLSafe(byte[] binaryData) {
-    return org.apache.commons.codec.binary.Base64.encodeBase64URLSafe(binaryData);
+    if(binaryData == null){
+      return null;
+    }
+
```
Comment (saicheems) [inline google-http-client/src/main/java/com/google/api/client/util/Base64.java]: nit: delete newline

File: google-http-client/src/main/java/com/google/api/client/util/Base64.java [hunk 3]
```diff
@@ -73,34 +79,43 @@ public static String encodeBase64String(byte[] binaryData) {
...
    * @param binaryData binary data to encode or {@code null} for {@code null} result
    * @return String containing Base64 characters or {@code null} for {@code null} input
-   * @see org.apache.commons.codec.binary.Base64#encodeBase64URLSafeString(byte[])
    */
   public static String encodeBase64URLSafeString(byte[] binaryData) {
-    return org.apache.commons.codec.binary.Base64.encodeBase64URLSafeString(binaryData);
+    if(binaryData == null){
+      return null;
+    }
+
```
Comment (saicheems) [inline google-http-client/src/main/java/com/google/api/client/util/Base64.java]: nit: delete newline

File: google-http-client/src/main/java/com/google/api/client/util/Base64.java (hunk 1 above)
Comment (garrettjonesgoogle) [reply google-http-client/src/main/java/com/google/api/client/util/Base64.java]: This is super major, I'm not sure what you're talking about :-P

File: google-http-client/src/main/java/com/google/api/client/util/Base64.java [hunk 4]
```diff
@@ -36,35 +36,41 @@
...
    * @return byte[] containing Base64 characters in their UTF-8 representation or {@code null} for
    *         {@code null} input
-   * @see org.apache.commons.codec.binary.Base64#encodeBase64(byte[])
    */
   public static byte[] encodeBase64(byte[] binaryData) {
-    return org.apache.commons.codec.binary.Base64.encodeBase64(binaryData);
+    if(binaryData == null){
+      return null;
+    }
+    return com.google.common.io.BaseEncoding.base64().encode(binaryData).getBytes();
```
Comment (garrettjonesgoogle) [inline google-http-client/src/main/java/com/google/api/client/util/Base64.java]: I don't think we need the fully qualified path any more (no more name conflicts)

File: google-http-client/src/test/java/com/google/api/client/util/Base64Test.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+import java.util.Arrays;
+
+/**
+ * Tests {@link Base64}.
+ * @author mwhisenhunt@google.com (Matt Whisenhunt)
+ */
+public class Base64Test extends TestCase {
+
+    public void testEmptyValue() {
```
Comment (garrettjonesgoogle) [inline google-http-client/src/test/java/com/google/api/client/util/Base64Test.java]: Use 2-space indentation

File: google-http-client/src/main/java/com/google/api/client/util/Base64.java [hunk 6]
```diff
@@ -36,35 +36,41 @@
...
   /**
    * Encodes binary data using the base64 algorithm but does not chunk the output.
    *
    * @param binaryData binary data to encode or {@code null} for {@code null} result
    * @return String containing Base64 characters or {@code null} for {@code null} input
-   * @see org.apache.commons.codec.binary.Base64#encodeBase64String(byte[])
    */
   public static String encodeBase64String(byte[] binaryData) {
-    return org.apache.commons.codec.binary.Base64.encodeBase64String(binaryData);
+    if(binaryData == null){
```
Comment (saicheems) [inline google-http-client/src/main/java/com/google/api/client/util/Base64.java]: oh and space between `if` and `(`, so:
`if (binaryData == null) {`

Comment (saicheems) [reply google-http-client/src/main/java/com/google/api/client/util/Base64.java]: (for all the if statements)

File: google-http-client/src/test/java/com/google/api/client/util/Base64Test.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+ */
+package com.google.api.client.util;
+
+import junit.framework.TestCase;
+
+import java.util.Arrays;
+
+/**
+ * Tests {@link Base64}.
+ * @author mwhisenhunt@google.com (Matt Whisenhunt)
```
Comment (saicheems) [inline google-http-client/src/test/java/com/google/api/client/util/Base64Test.java]: nit: newline above

File: google-http-client/src/main/java/com/google/api/client/util/Base64.java [hunk 8]
```diff
@@ -14,19 +14,13 @@
...
- * Proxy for version 1.6 (or newer) of the Apache Commons Codec
- * {@link org.apache.commons.codec.binary.Base64} implementation.
- *
- * <p>
- * This is needed in order to support platforms like Android which already include an older version
- * of the Apache Commons Codec (Android includes version 1.3). To avoid a dependency library
- * conflict, this library includes a reduced private copy of version 1.6 (or newer) of the Apache
- * Commons Codec (using a tool like jarjar).
- * </p>
+ * Convenient and now anachronistic proxy to 3rd party Base64 codec.
```
Comment (garrettjonesgoogle) [inline google-http-client/src/main/java/com/google/api/client/util/Base64.java]: Maybe instead: "Proxy for base 64 encoding/decoding which matches the Base64 interface in Apache Commons (for historical reasons)."

**Pull request merged**